doc/classes/ShapeCast2D.xml

```diff
@@ -0,0 +1,182 @@
+<?xml version="1.0" encoding="UTF-8" ?>
+<class name="ShapeCast2D" inherits="Node2D" version="4.0">
+	<brief_description>
+		Node for physics collision sweep and immediate overlap queries. Similar to the [RayCast2D] node.
+	</brief_description>
+	<description>
+		Shape casting allows to detect collision objects along the cast area (for instance, beam weapons).
+		Immediate collision overlaps can be easily done with the [member target_position] set to [code]Vector2(0, 0)[/code] and by calling [method force_shapecast_update] within the same [b]physics_frame[/b]. This also helps to overcome some limitations of [Area2D] when used as a continuos detection area, often requiring waiting a couple of frames before collision information is available to [Area2D] nodes, and when using the signals creates unnecessary complexity.
+		This node can detect multiple collision objects. If you want to use this just like [RayCast2D], use corresponding [code]get_closest_*[/code] methods instead.
+		[b]Note:[/b] shape casting is more computationally expensive compared to ray casting.
+	</description>
+	<tutorials>
+	</tutorials>
+	<methods>
+		<method name="add_exception">
+			<return type="void" />
+			<argument index="0" name="node" type="Object" />
+			<description>
+				Adds a collision exception so the shape does not report collisions with the specified node.
+			</description>
+		</method>
+		<method name="add_exception_rid">
+			<return type="void" />
+			<argument index="0" name="rid" type="RID" />
+			<description>
+				Adds a collision exception so the shape does not report collisions with the specified [RID].
+			</description>
+		</method>
+		<method name="clear_exceptions">
+			<return type="void" />
+			<description>
+				Removes all collision exceptions for this shape.
+			</description>
+		</method>
+		<method name="force_shapecast_update">
+			<return type="void" />
+			<description>
+				Updates the collision information for the shape. Use this method to update the collision information immediately instead of waiting for the next [code]_physics_process[/code] call, for example if the shape or its parent has changed state.
+				[b]Note:[/b] [code]enabled == true[/code] is not required for this to work.
+			</description>
+		</method>
+		<method name="get_closest_collider" qualifiers="const">
```

Comment:
The `get_closest_` functions are not going to return correct results until some changes are made in the physics server API, so better remove them for now (except for safe/unsafe distances).